Classes/DDTTYLogger.m

```diff
@@ -1247,7 +1247,7 @@ - (void)logMessage:(DDLogMessage *)logMessage {
             v[2].iov_base = (char *)msg;
             v[2].iov_len = msgLen;
 
-            if (_automaticallyAppendNewlineForCustomFormatters) {
+            if (iovec_len == 5) {
```

Comment:
It is not ignored, `_automaticallyAppendNewlineForCustomFormatters` was checked a few lines above the diff context shown here. `_automaticallyAppendNewlineForCustomFormatters` was used to decide how large the `io_vec` should be. 

Without this change, checking `_automaticallyAppendNewlineForCustomFormatters` again here causes a potential race condition if the property is changed from `@NO` to `@YES`.